pkg/importer/util.go

```diff
@@ -52,3 +54,72 @@ func StreamDataToFile(dataReader io.Reader, filePath string) error {
 	}
 	return nil
 }
+
+// A LimitedReader reads from R but limits the amount of
+// data returned to just N bytes. Each call to Read
+// updates N to reflect the new amount remaining.
+// Read returns EOF when N <= 0 or when the underlying R returns EOF.
+type LimitedReader struct {
+	R io.Reader // underlying reader
+	N int64     // max bytes remaining
+}
+
+func (l *LimitedReader) Read(p []byte) (n int, err error) {
+	if l.N <= 0 {
+		return 0, errors.New("EOF")
+	}
+	if int64(len(p)) > l.N {
+		p = p[0:l.N]
+	}
+	n, err = l.R.Read(p)
+	l.N -= int64(n)
+	return
+}
+
+// StreamDataToBlockDevice provides a function to stream the specified io.Reader to the specified block device
+func StreamDataToBlockDevice(dataReader io.Reader, filePath string) (written int64, err error) {
+	var fd int
+	fd, err = syscall.Open(filePath, syscall.O_CREAT|syscall.O_WRONLY, 0777)
+	defer syscall.Close(fd)
+	if err != nil {
+		return written, errors.Wrapf(err, "could not open block device %q", filePath)
+	}
+	glog.V(1).Infof("begin import...\n")
+	return copy(dataReader, fd)
+}
+
+func copy(dataReader io.Reader, fd int) (written int64, err error) {
```

Comment:
Looks like more code lifted from the golang `io` package.  Curious why you went the route of implementing your own copy instead of implementing your own `io.Writer` and using `io.Copy`?  Also, did you try wrapping the fd with `os.NewFile`?

Assuming we can't use `os.OpenFile` or `os.NewFile`, something like this should work and is cleaner IMO.

```golang
type MyWriter struct {
    F int
}

func (w *MyWriter) Write(b []byte) (int, error) {
    offset := 0
    total := len(b)
    for offset < total {
        n, err := Syscall.Write(w.fd, b[offset:total])
        offset += n
        if err != nil {
            return offset, err
        }
    }
    return offset, err
}

```